File: WorkManager/WorkManager.Data/Models/DataContext.cs
using WPFTools;
using WPFTools.Models;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using WorkManager.Data.Enums;

namespace WorkManager.Data.Models
{
    public class DataContext : EFContext
    {
        public DbSet<Project> Projects { get; set; }
        public DbSet<Task> Tasks { get; set; }
        public DbSet<TaskTime> TaskTimes { get; set; }

        public DbSet<Resource> Resources { get; set; }
        public DbSet<ProjectResource> ProjectResources { get; set; }
        public DbSet<TaskResource> TaskResources { get; set; }
        public DbSet<Team> Teams { get; set; }
        public DbSet<TeamAccount> TeamAccounts { get; set; }

        public DbQuery<V_Resource> V_Resources { get; set; }
        public DbQuery<V_Team> V_Teams { get; set; }
        public DbQuery<V_Project> V_Projects { get; set; }
        public DbQuery<V_Task> V_Tasks { get; set; }
        public DbQuery<EFValuesHistory> EFValuesHistory { get; set; }
        public DbQuery<V_ProjectStat> V_ProjectsStat { get; set; }
        public DbQuery<V_AccountStat> V_AccountStat { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<TaskTime>(entity =>
            {
                entity.ToTable("TASKTIMES");
                entity.Property(x => x.Type).HasConversion(x => (int)x, x => (TimeType)x);
            });
            modelBuilder.Entity<Task>(entity =>
            {
                entity.Property(x => x.State).HasConversion(x => (int)x, x => (TaskState)x);
            });
            modelBuilder.Query<V_ProjectStat>(entity =>
            {
                entity.Property(x => x.State).HasConversion(x => (int)x, x => (TaskState)x);
            });
            modelBuilder.Query<EFValuesHistory>(entity =>
            {
                entity.Property(x => x.ModType).HasConversion(x => (byte)x, x => (HistoryModificationType)x);
            });
            base.OnModelCreating(modelBuilder);
        }
        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            if (!optionsBuilder.IsConfigured)
            {
                optionsBuilder.UseSqlServer(Config.ConnectionString);
                optionsBuilder.UseQueryTrackingBehavior(QueryTrackingBehavior.NoTracking);
            }
            base.OnConfiguring(optionsBuilder);
        }
    }
}

File: WorkManager.Server/src/WorkManager/Data/Models/DataContext.java
package WorkManager.Data.Models;

import WorkManager.Data.QueryBuilders.DBTable;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class DataContext {
    public DataContext() throws SQLException {
        //String connectionUrl = "jdbc:sqlserver://DELL-02\\SQLSRV2017:1433;databaseName=WM;user=sa;password=sa;";
        String connectionUrl = "jdbc:sqlserver://PCV2\\SQLEXPRESS17:1433;databaseName=WM;user=sa;password=sa;";
        Connection connection = DriverManager.getConnection(connectionUrl);
        statement = connection.createStatement();
    }

    public static final DBTable Accounts = new DBTable("MGMNT","ACCOUNTS");
    public static final DBTable Projects = new DBTable("PROJECTS");
    public static final DBTable Tasks = new DBTable("TASKS");
    public static final DBTable TaskTimes = new DBTable("TASKTIMES");
    public static final DBTable Resources = new DBTable("RESOURCES");
    public static final DBTable ProjectResources = new DBTable("PROJECTRESOURCES");
    public static final DBTable TaskResources = new DBTable("TASKRESOURCES");
    public static final DBTable Teams = new DBTable("TEAMS");
    public static final DBTable TeamAccounts = new DBTable("TEAMACCOUNTS");

    public static final DBTable V_Resources = new DBTable("V_RESOURCES");
    public static final DBTable V_Teams = new DBTable("V_TEAMS");
    public static final DBTable V_Projects = new DBTable("V_PROJECTS");
    public static final DBTable V_Tasks = new DBTable("V_TASKS");
    public static final DBTable V_ProjectsStat = new DBTable("V_PROJECTSTAT");
    public static final DBTable V_AccountStat = new DBTable("V_ACCOUNTSTAT");
    public static final DBTable ValuesHistory = new DBTable("HST", "V_VALUESHISTORY");

    private Connection connection;
    private Statement statement;

    public Statement getStatement() { return statement; }
}
